### PR TITLE
Adding glslsandbox demos

### PR DESCRIPTION
**Pull request by MattCruikshank**
Let's see how the wall handles these!

### CODE REVIEW

File: demo_modules/glslsandbox/caramel.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    if (this.surface) {
+      this.surface.destroy();
+    }
+  }
+  
+  willBeShownSoon(container, deadline) {
+    const GlslSandboxSurface = require('client/surface/glslsandbox_surface');
+    
+    /*
+1. add this line
```
Comment (plural) [inline demo_modules/glslsandbox/caramel.js]: what's up with this commented code?

Comment (MattCruikshank) [reply demo_modules/glslsandbox/caramel.js]: They're my instructions to myself for how to port from glslsandbox...  I keep copying a demo module and modifying it, and these are the steps necessary.

**Comment by MattCruikshank** [comment]
And here's what they'll look like (roughly):

Caramel: http://glslsandbox.com/e#37985.0

Puls: http://glslsandbox.com/e#32147.0

Galaxy:  http://glslsandbox.com/e#41212.0

Red Swirls: http://glslsandbox.com/e#44398.0

Actually Puls is off-center vertically for some reason, and Red Swirls is mirrored about the X-axis right now, but that's the rough idea.

File: demo_modules/glslsandbox/caramel.js (hunk 1 above)
Comment (plural) [reply demo_modules/glslsandbox/caramel.js]: perhaps that is better as part of the surface file, then?  We had a site for docs, but I forgot where it lives.  :)

Comment (plural) [reply demo_modules/glslsandbox/caramel.js]: these are probably better off in the surface file, then.

**Comment by applmak** [comment]
I'm closing this CL because it's super stale.

**Pull request closed without merging**